frontend/src/metabase/models/containers/NewModelOptions/NewModelOptions.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useMemo } from "react";
+import { push } from "react-router-redux";
+import { t } from "ttag";
+import _ from "underscore";
+
+import { useMount } from "react-use";
+import { Location } from "history";
+import { Grid } from "metabase/components/Grid";
+import NewModelOption from "metabase/models/components/NewModelOption";
+
+import MetabaseSettings from "metabase/lib/settings";
+import * as Urls from "metabase/lib/urls";
+import Databases from "metabase/entities/databases";
+import { getHasDataAccess, getHasNativeWrite } from "metabase/selectors/data";
+
+import NoDatabasesEmptyState from "metabase/reference/databases/NoDatabasesEmptyState";
+
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { deserializeCardFromUrl } from "metabase/lib/card";
+import Database from "metabase-lib/metadata/Database";
+import {
+  OptionsGridItem,
+  OptionsRoot,
+  EducationalButton,
+} from "./NewModelOptions.styled";
+
+const EDUCATIONAL_LINK = MetabaseSettings.learnUrl("data-modeling/models");
+
+interface NewModelOptionsProps {
+  databases?: Database[];
+  location: Location;
+}
+
+const NewModelOptions = (props: NewModelOptionsProps) => {
+  const hasDataAccess = useSelector(() =>
+    getHasDataAccess(props.databases ?? []),
+  );
+  const hasNativeWrite = useSelector(() =>
+    getHasNativeWrite(props.databases ?? []),
+  );
+
+  const dispatch = useDispatch();
+
+  const collectionId = useMemo(() => {
+    if (location.hash) {
+      const decodedCollectionHash = deserializeCardFromUrl(
+        location.hash.replace(/^#/, ""),
+      );
+      return decodedCollectionHash.collectionId;
+    }
+    return null;
+  }, []);
+
+  useMount(() => {
+    const { location } = props;
+    if (location.search.length > 0) {
+      const { database, table, ...options } = location.query;
+      dispatch(
```

Comment:
This doesn't seem correct. I think we could pass `collectionId` when creating `NewModelOption` and that's it, we don't need to do this

Reply:
Gotcha, I found this logic a little weird when I first looked into this task. I can nuke this, add the collectionId logic, and make sure tests continue to pass. I don't see anywhere in the app where we pass `database` and `table` query parameters to the `/model/new` link